python/pyspark/sql/utils.py

```diff
@@ -24,6 +24,12 @@ class AnalysisException(Exception):
     """
 
 
+class IllegalArgumentException(Exception):
```

Comment:
Using IllegalArgumentException here will be more consistent with other API. Either one is good to me.